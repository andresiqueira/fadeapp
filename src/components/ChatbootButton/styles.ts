export const ChatbootButtonStyle = 
  `text-secondary
  fixed
  cursor-pointer 
  bg-primary 
  flex 
  items-center 
  justify-center 
  rounded-full 
  w-[2.938rem] 
  h-[2.938rem] 
  lg:w-[4.788rem] 
  lg:h-[4.788rem]  
  bottom-[5.375rem]
  right-[1.563rem] 
  lg:bottom-[3.563rem] 
  lg:right-[3.563rem]
  hover:bg-transparent
  hover:text-primary
  hover:border-primary
  hover:border-[0.063rem]
  transition ease-in-out
  focus:outline-none
  focus:ring
  focus:ring-0
  focus:ring-primary
  active:bg-transparent
  `

export const ChatbootIconStyle = `
  w-[1.116rem]
  h-[1.116rem]
  lg:w-[1.859rem]
  lg:h-[1.859rem]
`

export const ChatbootLabelStyle = `
  text-white
  text-xl
  fixed
  bottom-[8.675rem]
  right-[1.85rem]
  lg:bottom-[8.963rem] 
  lg:right-[4.9rem]
  leading-6
  font-medium 
`