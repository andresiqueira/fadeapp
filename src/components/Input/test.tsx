import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Input } from '.'
import { labelStyle } from "./styles"

const { getByLabelText, getByText } = screen

describe('<Input />', () => {
  it('Should render Input', () => {
    render(<Input label='nome' name='nome' />)
    const input = getByLabelText('nome')
    expect(input).toHaveAttribute('name', 'nome')
    expect(input).toHaveAttribute('id', 'nome')
    expect(input).toBeInTheDocument()
  })

  it('Should initial value is empty',() => {
     const initialValue = ''
    render(<Input label='nome' name='nome' value={initialValue} />)
    const input = getByLabelText('nome')
    const labelInput = getByText('nome')
    expect(labelInput).toHaveClass(labelStyle + "text-[1.125rem] leading-[1.3125rem] top-1")
    input.focus()
    expect(input).toHaveFocus()
    input.blur()
    expect(input).not.toHaveFocus()
  })

  it('Should change value on type', async () => {
    render(<Input label='nome' name='nome' value='' />)
    const name = 'Lucas Roberto'
    const input = getByLabelText('nome')
    userEvent.type(input, name)

    await waitFor(() => {
      expect(input).toHaveDisplayValue('Lucas Roberto')
      const labelInput = getByText('nome')
      expect(labelInput).toHaveClass(labelStyle + "text-[0.75rem] top-[-0.215rem] leading-[0.875rem]")
      input.focus()
      expect(input).toHaveFocus()
      input.blur()
      expect(input).not.toHaveFocus()
    })
  })

  it('Should match snapshot', () => {
    render(<Input label='nome' name='nome' value=''/>)
    const input = getByLabelText('nome')
    expect(input).toMatchSnapshot()
  })
})