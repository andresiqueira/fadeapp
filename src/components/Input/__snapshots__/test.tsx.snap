// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should match snapshot 1`] = `
<input
  autocomplete="off"
  class="
  box-border
  outline-0
  overflow-hidden
  w-full
  absolute
  rounded
  bg-transparent
  transition-all
  duration-200
  ease-in-out
  pt-[16px]
  pl-[0.5rem]
  text-[1rem]
  leading-[1.1875rem]
  text-secondary
  bottom-[0.4rem]
"
  id="nome"
  name="nome"
  value=""
/>
`;
